### PR TITLE
Merge NVC++ CMake changes into main

### PR DESCRIPTION
**Pull request by wmaxey**
These changes remove the need to use the upstream LLVM project on Windows for lit configuration.

### CODE REVIEW

**Comment by wmaxey** [comment]
This changes quite a bit of docker stuff, one thing I couldn't quite figure out was the default NVCC build where no architecture was specified. Prior to these changes it wasn't throwing up on tests that should have been unsupported.

To fix that behavior I changed config.py to launch with NVCC defaults and all newer SMs disabled.

File: .upstream-tests/utils/nvidia/linux/perform_tests.bash [hunk 1]
```diff
@@ -237,7 +239,7 @@ then
   ARCH_DETECTION_LOG=$(mktemp)
 
   LIBCUDACXX_SITE_CONFIG=${LIBCUDACXX_LIT_SITE_CONFIG} \
-  bash -c "lit -vv -a ${LIBCUDACXX_PATH}/test/nothing_to_do.pass.cpp" \
+  bash -c "lit -vv -a ${LIBCUDACXX_PATH}/test/nothing_to_do.pass.cpp -Dcompute_archs=${KNOWN_COMPUTE_ARCHS}" \
```
Comment (griwes) [inline .upstream-tests/utils/nvidia/linux/perform_tests.bash:242]: This expands into `-Dcompute_archs=35 50 52 53...`, no? Shouldn't it be `-Dcompute_archs='${KNOWN_COMPUTE_ARCHS}'`?

Comment (wmaxey) [reply .upstream-tests/utils/nvidia/linux/perform_tests.bash:242]: This probably explains some of the debugging I'm doing for older SMs. Nice catch.

Comment (wmaxey) [reply .upstream-tests/utils/nvidia/linux/perform_tests.bash]: Resolved with an amend.

**Review by griwes — APPROVED** [review]

File: .upstream-tests/utils/libcudacxx/test/config.py [hunk 2]
```diff
@@ -598,6 +598,12 @@ def configure_compile_flags(self):
                 self.config.available_features.add("pre-sm-70")
             if pre_sm_80:
                 self.config.available_features.add("pre-sm-80")
+        else : # Default assume that no features are available
```
[the remark this answers is not in the document]
Comment (griwes) [reply .upstream-tests/utils/libcudacxx/test/config.py]: Yeah.

**Pull request merged**